### ISSUE TITLE
Dataspace Versions

### ISSUE THREAD

**Issue by Aaron3154518** [BoiseState-AdaptLab/IEGenLib#102]
Every time a data space is written to, a new version should be created. This new version replaces the old version on every line between the last write and the current line.

**Comment by Aaron3154518** [comment]
@cathieO Do we need to add a statement for each new version which assigns the previous version's value to the new version?

**Comment by cathieO** [comment]
Nope. You need to replace the uses in the previous statements to the
new version and the read in the current statement.

On Tue, Jul 6, 2021 at 12:12 PM Aaron3154518 ***@***.***> wrote:
>
> @cathieO Do we need to add a statement for each new version which assigns the previous version's value to the new version?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.



-- 
==================================
 Catherine Olschanowsky, Ph.D.
 Assistant Professor
==================================

**Comment by rvishnkr** [comment]
SSA is not applied if a dataspace is not surrounded by `$` signs

**Comment by rvishnkr** [comment]
The dataspace get renamed to `dataSpaceName_number` when we encounter the first write. 
When we encounter any subsequent writes, the replaceDataSpaceName functions rechecks all the dataspaces names slowing it down significantly. We need to iterate backwards until we find the previous write based on the rename format `varName__w__number`

**Comment by rvishnkr** [comment]
@cathieO when we do renames how do we get the datatypes?
